### PR TITLE
Tatsu

### PR DESCRIPTION
**Pull request by ShioTatsu-Japan**

### CODE REVIEW

**Review by Kosei1227 — COMMENTED** [review]
Hi, it seems you just prepare the brief pipeine for setLeader and removeMember.
I believe setLeader and removeMember are the strong operation. So, unless the user is the current leader for the project, we should not set a new leader nor remove member. 
For this requirement, we need to have a method that checks whether the current user is the current leader.

**Comment by ShioTatsu-Japan** [comment]
I'm thinking of RemoveEmail to return an error when the member can’t be removed. Similarly, SetLeader will rise an error when old leader = new leader.

**Review by Kosei1227 — COMMENTED** [review]
Hi, I also modified firebaseaccessobject.
public FirebaseAccessObject() {
        try {
            // Check if FirebaseApp has already been initialized
            if (FirebaseApp.getApps().isEmpty()) {
                FileInputStream serviceAccount = new FileInputStream("Google_Firebase_SDK.json");
                FirebaseOptions options = new FirebaseOptions.Builder()
                        .setCredentials(GoogleCredentials.fromStream(serviceAccount))
                        .build();
                FirebaseApp.initializeApp(options);
            }
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
        this.db = FirestoreClient.getFirestore();
    }

By this way, I check whether firebase object is predefined or not.
So, basically, we do the same thing.
We can check whether both of them work when we run test cases after merge.

**Comment by makototakahara** [comment]
I read your javadocs, they are great. I appreciate the clarity and detail you've put into documenting each method. The class-level description provides a solid overview of the component's purpose.

This might be a very minor detail, but if I had to make a suggestion, I would capitalize the first word after @param (Parameter Name). For example, from "@param addEmailInputData the data transfer object containing information about the project and the email to be added" to "@param addEmailInputData The data transfer object containing information about the project and the email to be added". I think this may help developers understand better the format of your javadocs. 

Personally, as a beginner, this would have helped me read your javadocs a little bit easily. However, I am sure this is a trivial issue, especially for experienced developers used to the format of javadocs.

**Comment by ShioTatsu-Japan** [comment]
@makototakahara Thank you for your review! I capitalized the word after "@param" and "@return". I double checked, but please check again!

**Comment by Kosei1227** [comment]
When I run the main with "add email" version, I get a pop-up saying 
image
image
The top banner says "Create Project", but it should be "add email" or "remove email"

**Review by wogus00 — APPROVED** [review]
